File: java/Hot100/Hash/GroupAnagrams.java
package Hot100.Hash;

import java.util.*;

/**
 * Given an array of strings strs, group the anagrams together. You can return the answer in any order.
 */
public class GroupAnagrams {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String,List<String>> map=new HashMap<>();

        for(String str:strs){
            char[]chars=str.toCharArray();
            Arrays.sort(chars);
            String key=new String(chars);

            map.computeIfAbsent(key, k -> new ArrayList<>()).add(str);
        }

        return map.values().stream().toList();
    }
}

File: python/hash/group_anagrams.py
class Solution:
    def groupAnagrams(self, strs: list[str]) -> list[list[str]]:
        anagrams = {}

        for word in strs:
            # Sort the word to create a key
            sorted_word = "".join(sorted(word))
            if sorted_word not in anagrams:
                anagrams[sorted_word] = []
            anagrams[sorted_word].append(word)

        return list(anagrams.values())